File: Kotlin/isPrimeByForLoop.kt
fun main(){

    var isPrime = true

    print("Enter number: ")
    val num = readLine()!!.toInt()

    for (a in num-1 downTo 2 ){

        if (num%a==0){
            isPrime = false
            break
        }
    }
    if (isPrime){
        println("$num is a prime number")
    } else {
        println("$num is not a prime number")
    } 
}

File: CPP/isPrimeByForLoop.cpp

#include <iostream>
using namespace std;

int main() {
    
    int num,a;
    bool isPrime = true;
   
    cout << "Enter number: ";
    cin >> num;
    
    for(a=num-1; a>=2; a--){
        //printf("%d\n",a);
        if(num%a==0){
            isPrime = false;
            break;
        }
    }
    
    if(isPrime){
        printf("%d is a prime number",num);
    } else {
        printf("%d is not a prime number",num);
    } 
}
